locals.tf

```diff
@@ -66,8 +66,8 @@ locals {
               }
             }
             compactor = {
-              retention_delete_delay = "1h"
-              retention_enabled      = false
+              retention_delete_delay = "${var.compactor_retention.retention_delete_delay}"
```

Comment:
This only configures a delay before applying the retention. It doesn't configure the retention period.